src/components/admin/form-components/withSearchBar.jsx

```diff
@@ -0,0 +1,223 @@
+                      config.category
+                    }`}
+                    key={slug}
+                  >
+                    {/* eslint-disable react/jsx-no-bind */}
+                    <MenuItem
+                      leftIcon={<ContentAdd />}
+                      primaryText={this.state.searchValue}
+                      onClick={this.handleClick.bind(this, {
+                        value: this.state.searchValue,
+                        slug,
+                        id: null,
+                      })}
+                      disabled={this.props.disabled}
+                    />
+                    {/* eslint-enable react/jsx-no-bind */}
+                  </div>
+                )}
+              {/* eslint-disable react/jsx-no-bind */
+              this.state.searchSuggestions.map(item => (
+                <div
+                  className={`search-bar-result search-bar-${config.category}`}
+                  key={item.slug}
+                >
+                  <MenuItem
+                    primaryText={config.formatter(item)}
+                    onClick={this.handleClick.bind(this, item)}
+                    disabled={this.props.disabled}
+                  />
+                </div>
+              ))
+              /* eslint-enable react/jsx-no-bind */
+              }
+            </Menu>
+          </div>
+        </div>
+      );
+    }
+  }
+
+  /* eslint-disable consistent-return */
+  SearchBar.propTypes = {
+    model: React.PropTypes.shape({
+      get: React.PropTypes.func.isRequired,
+    }).isRequired,
+    fullWidth: React.PropTypes.bool,
+    length: React.PropTypes.number.isRequired,
+    handleClick: React.PropTypes.func.isRequired,
+    debounceTime: React.PropTypes.number,
+    disabled: React.PropTypes.bool,
+    enableAdd: React.PropTypes.bool,
+    // Each of these will be concatenated onto the base search path
+    // so don't supply the base search path
+    extraPathSets: (props, propName, componentName) => {
+      const prop = props[propName];
+      if (!prop) {
+        // Wasn't supplied so nevermind
+        return;
+      }
+      let error = new Error(
+        `Invalid prop ${propName} supplied to ${componentName}. It must be an array of pathSets.`,
+      );
+      if (!(prop instanceof Array)) {
+        return error;
+      }
+      const valid = prop.every(value => {
+        if (!(value instanceof Array)) {
+          return false;
+        } else if (value[0] === 'search') {
+          error = new Error(
+            `Invalid prop ${propName} supplied to ${componentName}. ` +
+              'Just add the extension, do not add "search"... as this is already considered.',
+          );
+          return false;
+        }
+        return true;
+      });
+      if (!valid) {
+        return error;
+      }
+    },
+  };
+
+  SearchBar.displayName = `SearchBar(${config.displayName})`;
+  return SearchBar;
+};
+
+export default withSearchBar;
```

Comment:
And just to enforce my new proposal of stopping default exports let's make this just `export withSearchBar`